axis_utils/include/axis_utils2.inc

```diff
@@ -425,45 +257,49 @@ end subroutine axis_edges
   !!   be the nearest data point to 0.0
   !!
   !!   @return real frac_index
-  function frac_index (value, array)
 
-    integer :: ia, i, ii, unit
-    real :: value !< arbitrary data...same units as elements in "array"
-    real :: frac_index
-    real, dimension(:) :: array !< array of data points  (must be monotonically increasing)
-    logical keep_going
 
+  function FRAC_INDEX_(value, array)
+
+    integer                               :: ia, i, ii, unit
+    real(kind=FMS_AU_KIND_)               :: value !< arbitrary data...same units as elements in "array"
+    real(kind=FMS_AU_KIND_)               :: FRAC_INDEX_
+    real(kind=FMS_AU_KIND_), dimension(:) :: array !< array of data points  (must be monotonically increasing)
+    logical                               :: keep_going
+    integer, parameter                    :: lkind = FMS_AU_KIND_
     ia = size(array(:))
 
-    do i=2,ia
+    do i = 2, ia
        if (array(i) < array(i-1)) then
           unit = stdout()
-          write (unit,*) &
-            '=> Error: "frac_index" array must be monotonically increasing when searching for nearest value to ', value
+          write (unit,*) '=> Error: "frac_index" array must be monotonically' &
+          & // 'increasing when searching for nearest value to ', value
           write (unit,*) '          array(i) < array(i-1) for i=',i
           write (unit,*) '          array(i) for i=1..ia follows:'
-          do ii=1,ia
+          do ii = 1, ia
              write (unit,*) 'i=',ii, ' array(i)=',array(ii)
           enddo
           call mpp_error(FATAL,' "frac_index" array must be monotonically increasing.')
        endif
     enddo
+
     if (value < array(1) .or. value > array(ia)) then
-!       if (value < array(1))  frac_index = 1.
-!       if (value > array(ia)) frac_index = float(ia)
-        frac_index = -1.0
+ !       if (value < array(1))  frac_index = 1.
+ !       if (value > array(ia)) frac_index = float(ia)
+        FRAC_INDEX_ = -1.0_lkind
     else
-       i=1
+       i = 1
        keep_going = .true.
        do while (i <= ia .and. keep_going)
           i = i+1
           if (value <= array(i)) then
-             frac_index = float(i-1) + (value-array(i-1))/(array(i)-array(i-1))
-             keep_going = .false.
+             FRAC_INDEX_ = real((i-1), FMS_AU_KIND_) + (real(value-array(i-1), FMS_AU_KIND_)) &
```

Comment:
do value and array need to be converted to FMS_AU_KIND_?

Reply:
No they do not. Will fix